### PR TITLE
[VIZ-15] Fixed implementation of #insert in Dataset and Unit Tested

### PR DESCRIPTION
**Pull request by as577**

### CODE REVIEW

File: concourse-plugin-core/src/main/java/com/cinchapi/concourse/plugin/data/Dataset.java [hunk 1]
```diff
@@ -193,7 +193,7 @@ public boolean insert(E entity, A attr, V value) {
             entities = Sets.newHashSet();
             index.put(value, entities);
         }
-        if(entities.add(entity)) {
+        if(index.get(value).add(entity)) {
```
Comment (jtnelson) [inline concourse-plugin-core/src/main/java/com/cinchapi/concourse/plugin/data/Dataset.java]: why'd you change this line?

Comment (as577) [inline concourse-plugin-core/src/main/java/com/cinchapi/concourse/plugin/data/Dataset.java]: I'm not exactly sure why, but this fixed the glitchy implementation. If you revert it back and run the Unit Test, it fails.

Comment (as577) [inline concourse-plugin-core/src/main/java/com/cinchapi/concourse/plugin/data/Dataset.java]: For some reason it wasn't adding it to the `Set<E>` that was associated with the map.

Comment (jtnelson) [inline concourse-plugin-core/src/main/java/com/cinchapi/concourse/plugin/data/Dataset.java]: oh, i know why...cool

**Pull request merged**